packages/core/content-releases/server/src/controllers/release-action.ts

```diff
@@ -17,6 +20,62 @@ const releaseActionController = {
       data: releaseAction,
     };
   },
+  async findMany(ctx: Koa.Context) {
+    const releaseId: GetReleaseActions.Request['params']['releaseId'] = ctx.params.releaseId;
+
+    const allowedContentTypes = getAllowedContentTypes({ strapi, userAbility: ctx.state.userAbility });
+    
+    const morphSanitizedPopulate = await allowedContentTypes.reduce(async (promiseAccumulator, contentTypeUid) => {
+      const accumulator = await promiseAccumulator;
+
+      const permissionsManager = await strapi.admin.services.permission.createPermissionsManager({
+        ability: ctx.state.userAbility,
+        model: contentTypeUid
+      });
+
+      // We create this to filter populated entries based on custom conditions on permissions
+      accumulator[contentTypeUid] = {
+        filters: permissionsManager.getQuery(CONTENT_MANAGER_READ_ACTION)
+      };
+
+      return accumulator;
+    }, Promise.resolve({} as Record<UID.ContentType, { filters: object }>));
```

Comment:
I'm not necessarily opposed to this, but I think there are some other options that might be nicer to read and maybe worth considering:
1. map the permission createPermissionsManager for each conteType, resolve them concurrently, reduce the resolved values to a new populate object
2. Use a for of loop and mutate an object